File: astra-ai-backend/src/main/java/com/astra/ai/auth/service/AuthService.java
package com.astra.ai.auth.service;

import com.astra.ai.dto.request.AdminRequest;
import com.astra.ai.dto.request.LoginRequest;
import com.astra.ai.dto.request.RegisterRequest;
import com.astra.ai.dto.response.LoginResponse;
import com.astra.ai.dto.response.RegisterResponse;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);
    LoginResponse login(LoginRequest request);
    RegisterResponse createAdmin(AdminRequest request);

}

File: astra-ai-ui/src/services/AuthService.ts
import api from "../api/api";
import type { ApiResponse } from "../types/api";
import type {
    LoginRequest,
    LoginResponse
} from "../types/auth";

class AuthService {

    async login(
        request: LoginRequest
    ): Promise<LoginResponse> {

        const response =
            await api.post<ApiResponse<LoginResponse>>(
                "/auth/login",
                request
            );

        return response.data.data;

    }

}

export default new AuthService();